### PR TITLE
[RFC] proposal to be able to download other files besides data/models

### PR DESCRIPTION
**Pull request by jerryjliu**
This proposal is to add an other_urls.csv section to be able to download any files that aren't explicitly related to a dataset and model. The motivating example here is the custom_metric.py file we are adding for regression. Also adjust the download script to be able to download other_urls.csv into an `other` folder

This proposal was mostly to address this comment https://github.com/RobustIntelligence/rime/pull/7551#pullrequestreview-1024191397. The other option was to just put custom metrics in the model urls. 

@RobustIntelligence/ml

### CODE REVIEW

**Comment by hinthornw** [comment]
I'm not sure I'll get around to reviewing the RFCs before release cut, sorry

**Comment by jerryjliu** [comment]
one downside of this approach is that in the SDK we only have functions called `upload_dataset_file` and `upload_model_directory`....which means we have to use one of these functions to upload the downloaded miscellaneous files. so not perfect

File: ri_public_examples/download_files.py [hunk 1]
```diff
@@ -10,21 +10,28 @@
 DEFAULT_CONFIG_PATH = "configs/stress_test_config.json"
 
 
-def _download_file(remote_path: str, local_path: Path) -> None:
+def _download_file(remote_path: str, local_path: Path) -> int:
     """Downloads file to local path."""
```
Comment (wzhang2022) [inline ri_public_examples/download_files.py]: nit: change to
"""Downloads file to local path and returns status code of request"""

File: ri_public_examples/download_files.py [hunk 2]
```diff
@@ -10,21 +10,28 @@
...
+) -> Optional[List[Tuple[str, Path]]]:
     """Fetch urls and output paths from text file."""
     # TODO: add data_urls/configs to package data directly
     # first download remote file path
     local_path = out_dir / Path(remote_file_path).name
     
-    _download_file(remote_file_path, local_path)
+    status_code = _download_file(remote_file_path, local_path)
+    if status_code != 200:
+        return None
```
Comment (wzhang2022) [inline ri_public_examples/download_files.py:41]: Should we raise some warning to the user if the download attempt fails?

**Comment by wzhang2022** [comment]
Other than that, LGTM

File: ri_public_examples/download_files.py [hunk 3]
```diff
@@ -4,28 +4,42 @@
...
 
 
 GITHUB_URL = "https://raw.githubusercontent.com/RobustIntelligence/ri-public-examples/master/"
 DEFAULT_CONFIG_PATH = "configs/stress_test_config.json"
 
 
-def _download_file(remote_path: str, local_path: Path) -> None:
-    """Downloads file to local path."""
+def _download_file(remote_path: str, local_path: Path) -> int:
+    """Downloads file to local path and returns status code of request.."""
```
Comment (wzhang2022) [inline ri_public_examples/download_files.py]: nit: probably typo, only use one period

File: ri_public_examples/download_files.py (hunk 2 above)
Comment (wzhang2022) [reply ri_public_examples/download_files.py:41]: Thumbs up as in we should, or thumbs up as in this is already fine and better as-is?

**Review by wzhang2022 — CHANGES_REQUESTED** [review]
LGTM again otherwise

File: ri_public_examples/download_files.py (hunk 2 above)
Comment (jerryjliu) [reply ri_public_examples/download_files.py:41]: i added a warning in download_file above

**Review by wzhang2022 — APPROVED** [review]
lgtm, seems like it can be merged once conflicts are resolved

**Pull request merged**